src/message.ts

```
@@ -36,6 +43,46 @@ export class Message {
         }
     }
 
+    /**
+     * nack - not ack for a message, meaning that the message will be redelivered again to the same consumers group without waiting to its ack wait time.
+     */
+    nack() {
+        if (this.message.nak)
+            this.message.nak();
+        else
+            throw MemphisError(new Error('cannot nack DLS message'));
+    }
+
+    /**
+     * deadLetter - Sending the message to the dead-letter station (DLS). the broker won't resend the message again to the same consumers group and will place the message inside the dead-letter station (DLS) with the given reason.
+     * The message will still be available to other consumer groups
+     * @param reason - the reason for the dead-lettering
+     * @returns void
+     */
+    deadLetter(reason: string) {
+        if (this._isInDls())
+            throw MemphisError(new Error('cannot dead-letter DLS message. message is already in DLS'));
```

ignore the error in this case